File: atcoder.jp/contests/abc198/tasks/abc198_b/main.cpp
#include <bits/stdc++.h>
using namespace std;

bool is_palindrome(const string &s) {
  string reversed = {s.rbegin(), s.rend()};
  return s == reversed;
}

string rstrip(const string &s, char c) {
  int n = s.size();
  for (int i = n - 1; i >= 0; i--) {
    if (s[i] != c) break;
    n--;
  }
  return s.substr(0, n);
}

int main() {
  string N;
  cin >> N;

  bool ans = is_palindrome(rstrip(N, '0'));
  cout << (ans ? "Yes" : "No") << endl;
}

File: atcoder.jp/contests/abc198/tasks/abc198_b/main.rb
# frozen_string_literal: true

def palindrome?(str = '')
  str.reverse == str
end

def rstrip(str = '', char = '')
  n = str.size
  (0...n).reverse_each do |i|
    break if str[i] != char

    n -= 1
  end
  str[0...n]
end

N = String gets.chomp

ans = palindrome? rstrip(N, '0')
puts ans ? 'Yes' : 'No'
